### ISSUE TITLE
Event does not propogate when subscribed IoC

### ISSUE THREAD

**Issue by NWoodsman** [thomaslevesque/WeakEvent#2]
I created a parent class with two children. Child 1 fires a WeakEvent. Child 2 has a public handler. In the _parent_ scope, we should be able to subscribe via inversion of control:

    public Parent()
    {
        Child1 = new Child1();
        Child2 = new Child2();
        Child1.Event += Child2.EventHandler;
    }

I have very weird behavior with the event firing off and on.  I have confirmed that a local handler inside Child1 is firing correctly. But Child2 doesn't fire as expected. It literally worked in one debug and failed in the next with no changes. Any ideas?

**Comment by thomaslevesque** [comment]
Hi @NWoodsman,

I'm not sure I understand what the problem is. What is `Child2.EventHandler`? Is it a method? A delegate field or property?
Could you give a bit more context? Maybe a short but complete repro?

**Comment by NWoodsman** [comment]
Sorry @thomaslevesque, I got back to the problem and it was my fault. `WeakEvent` works exactly as expected.

In the snippet I gave, `Child2.EventHandler` is a public method with the signature to handle the WeakEvent `Child1.Event`

I didn't realize that I had correctly written Child1 as a property of Parent, declared outside the scope of the Parent constructor. Whereas I was still declaring Child2 as a variable within the constructor. Once the constructor exited, WeakEvent was working _too well_ for my sleepy haze and letting Child2 get GC'ed. 

So I wasn't getting the event propogated because Child2 didn't exist. I moved Child2 to a property of Parent and all is good.

Thanks, I'll close this one.
